--- Makefile.in.orig    2017-11-11 17:22:13.612000000 +0000                                                                             
+++ Makefile.in 2017-11-11 17:21:31.480000000 +0000                                                                                     
@@ -533,6 +533,8 @@                                                                                                                     
        $(INSTALL) -m 755 -d $(DESTDIR)$(mandir)/man8                                                                                   
        $(INSTALL) -m 755 -d $(DESTDIR)$(mandir)/man5                                                                                   
        $(INSTALL) -m 755 -d $(DESTDIR)$(mandir)/man1                                                                                   
+       $(INSTALL) -m 644 -d $(DESTDIR)$(libdir)/pkgconfig                                                                              
+       $(INSTALL) -m 644 contrib/libunbound.pc $(DESTDIR)$(libdir)/pkgconfig                                                           
        $(LIBTOOL) --mode=install cp -f unbound$(EXEEXT) $(DESTDIR)$(sbindir)/unbound$(EXEEXT)                                          
        $(LIBTOOL) --mode=install cp -f unbound-checkconf$(EXEEXT) $(DESTDIR)$(sbindir)/unbound-checkconf$(EXEEXT)
        $(LIBTOOL) --mode=install cp -f unbound-control$(EXEEXT) $(DESTDIR)$(sbindir)/unbound-control$(EXEEXT)
