--- old/compat/getentropy_linux.c	2018-07-22 12:34:59.000000000 +0200
+++ new/compat/getentropy_linux.c	2018-07-22 12:35:57.000000000 +0200
@@ -56,9 +56,6 @@
 #include <linux/types.h>
 #include <linux/random.h>
 #include <linux/sysctl.h>
-#ifdef HAVE_GETAUXVAL
-#include <sys/auxv.h>
-#endif
 #include <sys/vfs.h>
 #ifndef MAP_ANON
 #define MAP_ANON MAP_ANONYMOUS
@@ -527,24 +524,6 @@
 
 			HD(cnt);
 		}
-#ifdef HAVE_GETAUXVAL
-#  ifdef AT_RANDOM
-		/* Not as random as you think but we take what we are given */
-		p = (char *) getauxval(AT_RANDOM);
-		if (p)
-			HR(p, 16);
-#  endif
-#  ifdef AT_SYSINFO_EHDR
-		p = (char *) getauxval(AT_SYSINFO_EHDR);
-		if (p)
-			HR(p, pgs);
-#  endif
-#  ifdef AT_BASE
-		p = (char *) getauxval(AT_BASE);
-		if (p)
-			HD(p);
-#  endif
-#endif /* HAVE_GETAUXVAL */
 
 		CRYPTO_SHA512_FINAL(results, &ctx);
 		memcpy((char*)buf + i, results, min(sizeof(results), len - i));
