--- a/configure
+++ b/configure
@@ -22234,8 +22234,7 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to enable Secure Transport" >&5
 $as_echo_n "checking whether to enable Secure Transport... " >&6; }
 if test -z "$ssl_backends" -o "x$OPT_SECURETRANSPORT" != xno; then
-  if test "x$OPT_SECURETRANSPORT" != "xno" &&
-     test -d "/System/Library/Frameworks/Security.framework"; then
+  if test "x$OPT_SECURETRANSPORT" != "xno"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }

@@ -21647,7 +21647,7 @@
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for gdi32" >&5
 $as_echo_n "checking for gdi32... " >&6; }
       my_ac_save_LIBS=$LIBS
-      LIBS="-lgdi32 $LIBS"
+      LIBS="$LIBS -lgdi32"
       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <windef.h>
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -55,7 +55,7 @@ include Makefile.inc
 curl_SOURCES = $(CURL_FILES)

 # This might hold -Werror
-CFLAGS += @CURL_CFLAG_EXTRAS@
+CFLAGS += @CURL_CFLAG_EXTRAS@ -DBUILDING_LIBCURL

 # Prevent LIBS from being used for all link targets
 LIBS = $(BLANK_AT_MAKETIME)
