diff --git a/Make.Rules b/Make.Rules
index fed00c9..ca58f92 100644
--- a/Make.Rules
+++ b/Make.Rules
@@ -49,10 +49,10 @@ MINOR=36
 KERNEL_HEADERS := $(topdir)/libcap/include/uapi
 IPATH += -fPIC -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 
-CC := $(CROSS_COMPILE)gcc
+# CC := $(CROSS_COMPILE)gcc
 DEFINES := -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 CFLAGS := -O2 $(DEFINES)
-BUILD_CC := $(CC)
+BUILD_CC := $(CROSS_COMPILE)gcc
 BUILD_CFLAGS := -O2 $(DEFINES) $(IPATH)
 AR := $(CROSS_COMPILE)ar
 RANLIB := $(CROSS_COMPILE)ranlib
